src/client.js

```diff
@@ -1,27 +1,56 @@
 "use strict";
 
-const net = require("net");
-const url = require("url");
+const net = require("net")
+const url = require("url")
+const {dialog} = require("electron")
 
 // Client can read/write messages from a TCP server
 class Client {
-  // init initializes the Client
-  init(addr) {
-    let u = url.parse("tcp://" + addr, false, false);
-    this.socket = new net.Socket();
-    this.socket.connect(u.port, u.hostname, function() {});
-    this.socket.on("close", function() {
-      process.exit();
-    });
-    return this;
-  }
-
-  // write writes an event to the server
-  write(targetID, eventName, payload) {
-    let data = { name: eventName, targetID: targetID };
-    if (typeof payload !== "undefined") Object.assign(data, payload);
-    this.socket.write(JSON.stringify(data) + "\n");
-  }
+    // init initializes the Client
+    init(addr) {
+        this.connect(addr)
+
+        this.socket.on('error', function(err){
+            // Raising an exception in case of any error in socket
+            const messageBoxOptions = {
```

Comment:
Why do you need to show the error in a dialog ? Writing it in `stdout` or `stderr` should be sufficient since it should be picked up by the `go` side.

Reply:
I thought showing the message in a proper window is what you suggested in your earlier comment, so I made the changes accordingly plus showing a proper reason as to why the application closed/crashed abruptly will be a better option.

Reply:
OK makes sense. Once the last issues are resolved, I'll test this PR out and see if showing a dialog is actually the best option.

Reply:
@asticode please test this PR

> OK makes sense. Once the last issues are resolved, I'll test this PR out and see if showing a dialog is actually the best option.